Mathlib/RingTheory/TensorProduct.lean

```diff
@@ -574,26 +574,21 @@ variable [Semiring D] [Algebra R D]
 and evidence of multiplicativity on pure tensors.
 -/
 def algHomOfLinearMapTensorProduct (f : A ⊗[R] B →ₗ[S] C)
-    (w₁ : ∀ (a₁ a₂ : A) (b₁ b₂ : B), f ((a₁ * a₂) ⊗ₜ (b₁ * b₂)) = f (a₁ ⊗ₜ b₁) * f (a₂ ⊗ₜ b₂))
-    (w₂ : ∀ r, f (algebraMap S A r ⊗ₜ[R] 1) = algebraMap S C r) : A ⊗[R] B →ₐ[S] C :=
-  { f with
-    map_one' := by rw [← (algebraMap S C).map_one, ← w₂, (algebraMap S A).map_one]; rfl
-    map_zero' := by simp only; rw [LinearMap.toFun_eq_coe, map_zero]
-    map_mul' := fun x y => by
+    (map_mul : ∀ (a₁ a₂ : A) (b₁ b₂ : B), f ((a₁ * a₂) ⊗ₜ (b₁ * b₂)) = f (a₁ ⊗ₜ b₁) * f (a₂ ⊗ₜ b₂))
+    (map_one : f (1 ⊗ₜ[R] 1) = 1) : A ⊗[R] B →ₐ[S] C :=
```

Comment:
I agree that `w₁` and `w₂` were terrible names, but I don't like `map_mul` and `map_one` either since they shadow the corresponding lemmas, and then it's confusing reading the source. Can we call them `h_map_mul` and `h_map_one`, or `h_mul` and `h_one`?

Also, can you fix the docstring. Is it suppose to say "from a *nonzero* map" or something to reflect the `f 1 = 1` condition?

Aside: is there a reason not to just write `f 1 = 1` in the hypothesis? I'm okay either way.

Reply:
>  Can we call them ...  
> Also, can you fix the docstring? 

Done

> Aside: is there a reason not to just write f 1 = 1 in the hypothesis? I'm okay either way.

Yes, I explained this in the docstring